### DISCUSSION TITLE
why indexing so slow?

### DISCUSSION THREAD

**Discussion by Wenze7** [castorini/pyserini#1371] [Q&A] [answered]
I use the following script to index
<img width="707" alt="image" src="https://user-images.githubusercontent.com/55132749/205556165-86d675b8-8e01-40e8-bd60-f7290c24e291.png">
I have 160G jsonl file, and the number of thread is 24, but the speed is so slow, the following task have ran 5 hour, but only 300K docs has been processed. I want to ask how to speed?
<img width="1176" alt="image" src="https://user-images.githubusercontent.com/55132749/205556632-b581d4d9-aab3-4ea4-bb84-fe82ce38c3ca.png">

**Comment by lintool** [comment, marked as answer]
How many files do you have? With 24 threads, you need to have at least 24 _separate_ files to take advantage of the parallelism, i.e., each thread works on a file independently.

**Reply by Wenze7** [reply]
Thanks! And i have some new problems, I put many *.jsonl file in one dir to index instead of *.jsonl.gz, I'm not sure if it can work?
And in the following log, why so many lines to show 0 documents indexed?
<img width="1184" alt="image" src="https://user-images.githubusercontent.com/55132749/205689488-05350428-5f40-48e5-bba1-6251f28d6e8e.png">

**Reply by lintool** [reply]
No, what you see is normal. The indexer does not update stats until it's completed _files_.

**Reply by Wenze7** [reply]
Thanks, I put many *.jsonl file in one dir to index instead of *.jsonl.gz, I'm not sure if it can work?

**Reply by lintool** [reply]
Yes, the indexer will handle both `.jsonl` and `.jsonl.gz` files. Compression is preferred to save space.
